users/dierkes/preprocessing/wav2vec.py

```diff
@@ -0,0 +1,67 @@
+import soundfile
+import glob
+import os
+import random
+
+from sisyphus import Job, Task, tk
+
+class FairseqAudioManifestCreationJob(Job):
+    """
+    Creates required manifest files for wav2vec pretraining with fairseq. For the original
+    facebook script consider https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/wav2vec_manifest.py
+    """
+    def __init__(
+        self, audio_dir_path, file_extension="wav", valid_percent=0.01, seed=42, path_must_contain=None
+    ):
+        """
+        :param tk.Path audio_dir_path: path to raw audio files to be included
+        :param str file_extension: file extension to look for in audio_dir_path
+        :param float valid_percent: percentage of files to be in validation set
+        :param int seed: random seed for splitting into train and valid set
+        :param str|None path_must_contain: if set, path must contain this substring
+            for a file to be included in the manifest
+        """
+        self.audio_dir_path = audio_dir_path
+        self.file_extension = file_extension
+        self.valid_percent = valid_percent
+        assert 0 <= self.valid_percent <= 1.0
+        self.seed = seed
+        self.path_must_contain = path_must_contain
+
+        self.out_manifest_path = self.output_path("manifest/", directory=True)
+        self.rqmt = {"time": 8, "mem": 8, "cpu": 1}
```

Comment:
When I ran the job it took about ~45 minutes for a medium sized corpora (e.g. german hykist corpus). Regarding the computational requirements I think it can be a mini task.

Reply:
Oh okay, that's much longer than I expected. Then maybe reduce `time` to 2 here and don't make it a mini task.